Sources/DCoreKit/Model/NftModel.swift

```diff
@@ -0,0 +1,130 @@
+                guard let name = property.label, let anyValue = try convertToAnyValue(property.value)
+                else { return nil }
+                return (name.dropFirst_(), anyValue)
+            }
+        .map { Pair($0.0, $0.1) }
+    }
+
+    func createDefinitions() -> [NftDataType] {
+        return Mirror(reflecting: self).children.map { child in
+            switch child.value {
+            case let property as NftProperty<String>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<Int>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<UInt8>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<UInt32>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<UInt64>: return property.asNftDataType(name: child.label)
+            case let property as NftProperty<Bool>: return property.asNftDataType(name: child.label)
+            default:
+                return NftDataType(
+                    type: getTypeLiteral(value: child.value), unique: false, modifiable: .nobody, name: child.label
+                )
+            }
+        }
+    }
+}
+
+private func getTypeLiteral(value: Any) -> NftDataType.TypeLiteral {
+    switch value {
+    case is Int, is UInt8, is UInt32, is UInt64: return .integer
+    case is Bool: return .boolean
+    case is String: return .string
+    default: preconditionFailure("Unsupported type \(value.self)")
+    }
+}
+
+private extension NftProperty {
+    func asNftDataType(name: String?) -> NftDataType {
+        return NftDataType(
+            type: getTypeLiteral(value: wrappedValue),
+            unique: unique,
+            modifiable: modifiableBy,
+            name: name?.dropFirst_()
+        )
+    }
+}
+
+/**
+ RawNft data model that only contains values of properties.
+ This model should be converted to a specific `NftModel` that contains specification for each property
+ */
+public struct RawNft: NftModel {
+    public let values: [AnyValue]
+
+    public init() {
+        values = []
+    }
+
+    public init(from decoder: Decoder) throws {
+        self.values = try decoder.singleValueContainer().decode([AnyValue].self)
+    }
+
+    /**
+     Binds values from RawNft to properties of NftModel specified by generic parameter.
+     Binding is based on order of parameters specified in the source code of the model.
+
+     Returns: Instance of specified type conforming to `NftModel`
+     */
+    public func toNftModel<T: NftModel>() throws -> T {
+        let mirror = Mirror(reflecting: T())
+        let pairs = mirror.children.enumerated().map {
+            ($0.element.label?.dropFirst_() ?? "", values[$0.offset])
+        }
+        let dictionary = pairs.reduce(into: [:]) { $0[$1.0] = $1.1 }
+        let anyValue = AnyValue.object(dictionary)
+        let encoded = try JSONEncoder().encode(anyValue)
```

Comment:
JSONEncode or JSONDecoder should not use codingContext()?

Reply:
this is used purely for internal purposes of working around Swift's lack of proper reflection API, so that we can build instance of unknown type from key-value pairs from DCore. As such we don't need to support BigInt or other common settings for the json decoder that are meant to be used for communication with DCore.